src/main/java/net/wimpi/modbus/net/TCPMasterConnection.java

```diff
@@ -233,7 +225,9 @@ public void resetConnection() {
 
     @Override
     public String toString() {
-        return new ToStringBuilder(this, toStringStyle).append("socket", m_Socket).toString();
+        return "TCPMasterConnection [m_Socket=" + m_Socket + ", m_Timeout=" + m_Timeout + ", m_Connected=" + m_Connected
```

Comment:
Risk of NPE? E.g. Socket can be null

Reply:
Yes, I suppose you are right.

Reply:
No. Actually I was right after all. 

The function of Java string concatenation (like `z= a + b + c`) is really clever; it does the following things implicitly:
1. If an argument is a class other than String then Java expands `argument` to `argument.toString()`
2. If an argument is `null` then Java does not use it in the resulting string; and it does not create an NPE

Try the following code..
```
    @Test
    void test() {
        String a = "the" + "quick" + null + "brown";
        assertEquals(a, "thequickbrown");
    }
```

Reply:
Thanks, I learned something new 🤯

Reply:
Me too. Your comment was the trigger for me to look it up.